src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -769,6 +770,37 @@ internal static object Create(IHostEnvironment env, ModelLoadContext ctx, IHost
 
             return genericInstance;
         }
+
+        internal static Type GetLoadType(Type modelType)
+        {
+            var att = modelType.GetCustomAttribute(typeof(PredictionTransformerLoadType)) as PredictionTransformerLoadType;
+            if (att != null)
+            {
+                if (att.LoadType.IsGenericType && att.LoadType.GetGenericArguments().Length == modelType.GetGenericArguments().Length)
+                {
+                    // This assumes that if att.LoadType and modelType have the same number of type parameters
+                    // Then they should get the same type parameters.
+                    // This is the case for CalibratedModelParametersBase and its children generic clases.
+                    // But might break if other classes begin using the PredictionTransformerLoadtype attribute in the future.
+                    Type[] typeArguments = modelType.GetGenericArguments();
+                    Type genericType = att.LoadType;
+                    return genericType.MakeGenericType(typeArguments);
+                }
+            }
+
+            return modelType;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    internal class PredictionTransformerLoadType : Attribute
```

Comment:
Attribute classes should have the suffix `Attribute`. So this should be changed to `internal class PredictionTransformerLoadTypeAttribute : Attribute`